### PR TITLE
Remove vestigial debug print statement in `walk_head_nodes`

### PR DESCRIPTION
**Pull request by shadeMe**
## Description

Removes one of the last, potential GIL-Refnanny related regressions.

### Types of change

Bug fix

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
I realize that it's a really small change, but all the reformatting makes this PR kind of hard to follow and is going to make the history really confusing in the future. (I had to use a text search to find the place where the print statement was removed in the commit.)

We don't have automatic processes for formatting cython, so if there's a good solution that you're aware of, it would be useful to add it. If it's not, then I would prefer if this kind of PR didn't do any unrelated reformatting, or at least not beyond a few lines at most. What does everyone think?

**Comment by shadeMe** [comment]
Ah, I didn't notice all the whitespace changes on my end. I can revert them, not an issue. That said, the GitHub diff interface does let one hide whitespace changes: Click on the gear icon at the top of the _Files Changed_ page, check "Hide whitespace" and hit _Apply_.

I don't have a specific formatter for Cython, but my editor is configured to automatically trim trailing whitespaces (which is what has happened here).

**Comment by svlandeg** [comment]
> I realize that it's a really small change, but all the reformatting makes this PR kind of hard to follow and is going to make the history really confusing in the future. (I had to use a text search to find the place where the print statement was removed in the commit.)
> 
> We don't have automatic processes for formatting cython, so if there's a good solution that you're aware of, it would be useful to add it. If it's not, then I would prefer if this kind of PR didn't do any unrelated reformatting, or at least not beyond a few lines at most. What does everyone think?

Agreed! As long as we don't have a standard way of formatting Cython, I'd avoid making too many formatting changes in parts of the code that would otherwise be untouched by a PR. Mostly for `git blame`'s sake ;-)

**Review by svlandeg — COMMENTED** [review]
The `graph.pyx` stuff is not currently used by anyone or anything, as far as I know. There's actually plenty of other `print` statements that should be cleaned up in this same file, or at least put behind some `debug` variable or something.

**Comment by shadeMe** [comment]
I only see two other `print` statements in the `graph.pyx` file on my end. Both of them are in regular Python functions, so they shouldn't concern us at least with respect to RefNanny. Still, I could remove them in this PR if that's what we want to do (feature-gating them seems like overkill in this case given how ad-hoc they appear to be).

In any event, I do think it makes sense to merge the proposed change as it will eliminate a potential bottleneck in the future, especially given that the GIL lock was being acquired in a graph traversal method.

**Comment by svlandeg** [comment]
Yea, let's just remove the other `print` statements too (you're right that there are currently 3 in total, I seemed to recall there were more)
